File: calculation/src/test/java/com/rustamg/calc/loan/utils/GsonFactory.java
package com.rustamg.calc.loan.utils;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;


/**
 * Created by rustamg on 29/07/15.
 */
public class GsonFactory {

    private static final String YYYY_MM_DD = "yyyy-MM-dd";

    public static Gson create() {

        return new GsonBuilder()
                .serializeSpecialFloatingPointValues()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                .setDateFormat(YYYY_MM_DD)
                .create();
    }
}

File: core/data/src/main/java/ru/kredit/calculator/data/util/GsonFactory.kt
package ru.kredit.calculator.data.util

import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder

object GsonFactory {
    private const val DATE_FORMAT = "yyyy-MM-dd"

    fun create(): Gson {
        return GsonBuilder()
            .serializeSpecialFloatingPointValues()
            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
            .setDateFormat(DATE_FORMAT)
            .create()
    }
}
